frontend/src/OneBuilding/index.jsx

```diff
@@ -0,0 +1,189 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  Text, View, SafeAreaView, StyleSheet,
+  Modal, ActivityIndicator, TouchableOpacity, TouchableWithoutFeedback,
+} from 'react-native';
+import PropTypes from 'prop-types';
+
+import ModalHeader from './ModalHeader';
+
+import { fetchOneBuilding } from './actionCreator';
+import PopupItem from './popupItem';
+import colors from '../common/colors';
+
+import { fetchTroops } from '../Troops/actionCreator';
+
+import TownhallDetail from './TownhallDetail';
+import AcademyDetail from './AcademyDetail';
+import FarmDetail from './FarmDetail';
+import MineDetail from './MineDetail';
+
+const styles = StyleSheet.create({
+  background: {
+    flex: 1,
+    backgroundColor: 'rgba(0,0,0,0.6)',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  container: {
+    width: 300,
+    backgroundColor: 'rgb(202,202,202)',
+    borderRadius: 6,
+  },
+  mainBody: {
+    padding: 15,
+    paddingTop: 0,
+  },
+  loading: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+});
+
+function getDetailInfo(
+  type,
+  troops,
+  gold,
+  food,
+  goldGenerateRate,
+  foodGenerateRate,
+  buildingLevel,
+  createTroopGoldCost,
+  createTroopTimeCost,
+  upgradeAcademyGoldCost,
+  upgradeAcademyTimeCost,
```

Comment:
if a function needs this much params, it should be splitted up somehow into smaller parts